docs: Extend Hubble Performance Tuning Documentation (#34665)
cilium-team/cilium: Documentation/operations/performance/tuning.rst

Increase Hubble Event Queue Size
--------------------------------

The Hubble Event Queue buffers events after they have been emitted from datapath and
before they are processed by the Hubble subsystem. If this queue is full, because Hubble
can't keep up with the amount of emitted events, Cilium will start dropping events.
This does not impact traffic, but the events won't be processed by Hubble and won't show
up in Hubble flows or metrics.

When this happens you will see log lines similar to the following.

::

   level=info msg="hubble events queue is processing messages again: NN messages were lost" subsys=hubble
   level=warning msg="hubble events queue is full: dropping messages; consider increasing the queue size (hubble-event-queue-size) or provisioning more CPU" subsys=hubble

By default the Hubble event queue size is ``#CPU * 1024``, or ``16384`` if your nodes have
more than 16 CPU cores. If you encounter event bursts that result in dropped events,
increasing this queue size might help. We recommend gradually doubling the queue length
until the drops disappear. If you don't see any improvements after increasing the queue

> length to 128k, further increasing the event queue size unlikely to help.

Reviewer: ```suggestion
length to 128k, further increasing the event queue size is unlikely to help.
```

Be aware that increasing the Hubble event queue size will result in increased memory

> usage. Depending on your traffic pattern, increasing the queue size by ``10'000`` may

Reviewer: ```suggestion
usage. Depending on your traffic pattern, increasing the queue size by ``10,000`` may
```

Increasing the Hubble event queue size can't mitigate a consistently high rate of events
being emitted by Cilium datapath and it does not reduce CPU utilization. For this you
should consider increasing the aggregation interval or rate limiting events.

Increase Aggregation Interval
-----------------------------

By default Cilium generates a tracing event on every new connection, any time a packet
contains TCP flags that have not been previously seen for the packet direction, and on
average once per ``monitor-aggregation-interval``, which defaults to 5 seconds.

Depending on your network traffic patterns, the re-emitting of trace events per
aggregation interval can make up a large part of the total events. Increasing the
aggregation interval may decrease CPU utilization and can prevent lost events.

The following will set the aggregation interval to 10 seconds.

.. tabs::
    .. group-tab:: Cilium CLI

       .. parsed-literal::

           cilium install |CHART_VERSION| \\
             --set bpf.events.monitorInterval="10s"

    .. group-tab:: Helm

       .. parsed-literal::

           helm install cilium |CHART_RELEASE| \\
             --namespace kube-system \\

>              --set bpf.events.monitorInterval="10s"

Reviewer 2: "bpf.**events**.monitorInterval"
and
"bpf.**events**.monitorInterval"
seems not existing. Is it a regression? I think the key `events` is obsolete and the interval can be controlled directly via `bpf.monitorInterval`: https://github.com/search?q=repo%3Acilium%2Fcilium%20%22bpf.monitorInterval%22&type=code

             --set bpf.events.default.rateLimit=10000 \\
             --set bpf.events.default.burstLimit=50000

> 
> You can also choose to stop exposing event types in which you

Reviewer: No need for the additional blank line

```suggestion
You can also choose to stop exposing event types in which you
```
